libraries/ms-helper/inc/thermistor_converter.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+// Thermistor Reading interface - converts an ADC voltage reading to temperature
+// Requires GPIO and Interrupts to be initialized.
+#include "adc.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "soft_timer.h"
+
+#define RESISTANCE_LEN 71
+
+typedef struct {
+  uint32_t sibling_resistance;  // milliohms
+  uint32_t source_voltage;      // millivolts
+  void *context;
+
+  ADCChannel adc_channel;
+} ThermistorStorage;
+
+typedef struct {
+  uint32_t sibling_resistance;  // milliohms
+  uint32_t source_voltage;      // millivolts
```

Comment:
You should be reading the reference voltage from the ADC since you're getting data from the ADC anyways.

Reply:
The source_voltage isn't the value we're reading from the ADC but the initial voltage before the current crosses the thermistor. Since we need to know the voltage drop across the thermistor we need the intial value (unless there is some other way to get that from the ADC?).

Reply:
Yes, the thermistor's supply voltage should be that of the ADC's. You can use the reference channel for that.

Reply:
This function
uint16_t prv_get_vdda(uint16_t reading) ?

Reply:
No, you shouldn't be using internal functions. Look at the header.